watcher.py

```diff
@@ -96,7 +125,7 @@ def run(self):
             self.results[test]['subtest'][subtest]['data'] = result
 
         self.log('Cleaning up recordings...')
-        for test, result in self.results.iteritems():
+        for test, result in list(self.results.iteritems()):
```

Comment:
The code in the for loop modifies self.results, which breaks if an iterator is in use.